### DISCUSSION TITLE
Problem with nuxt router

### DISCUSSION THREAD

**Discussion by kylawamg** [nuxt/nuxt#9806] [Questions (Nuxt 2)]
Hi, 

I not know how to achieve the following problem and I hope someone could help me. 

Im migrating a test.vue that renders one component or another depending on a step property on the route path. So for example, by default, the test.vue page renders component1.vue and when the user makes some action I add ?step=1 to the route in order to render other component inside test.vue but when I tried it in nuxt I get a 404 error because I dont know how to config the router properly or the file structure...

The number of possible steps is limited to 2 so I have 3 posibilities:  

/test: Load the page with the default component.
/test?step=1: I want to render the same page but different content with v-if based in the step property
/test?step=2: Same, I want to render test.vue but with different content based in the step property.

Thanks for the support in advance!

**Comment by atinux** [comment]
Do you have a reproduction?

You can use https://template.nuxtjs.org

**Comment by kissu** [comment]
You'll need to make some condition depending on the router. You should give a read to dynamic components: https://vuejs.org/v2/guide/components-dynamic-async.html#keep-alive-with-Dynamic-Components

You don't even need to mount/unmount new components, just make it dynamic. 👌🏻
